### DISCUSSION TITLE
posts/Go/read_gin_sourcecode/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#335] [Announcements]
https://www.liwenzhou.com/posts/Go/read_gin_sourcecode/ 

 李文周的Blog gin 框架 go web 源码 路由 中间件 分析 trie radix tree 对象池 sync.Pool c.Next c.Abort

**Comment by Anaethesia** [comment]
从头到尾看完大佬的gin讲解，通俗易懂，也加深了自己对web开发的理解，十分感谢。

**Comment by sodaling** [comment]
这边补充一下，root tree 用slice而不是用map并不是为了内存考虑的。在官方issue有人提问过这个问题，答案是在短长度的情况下，slice的速度会比map快很多
https://github.com/gin-gonic/gin/issues/2320
